### ISSUE TITLE
pnp-filepreview image onerror not loading

### ISSUE THREAD

**Issue by FredrikEkstroem** [microsoft-search/pnp-modern-search#1376]
**Version used**
Ex: 4.3.0

**Describe the bug**
If the preview image doesn't exist the alternate preview image is not loaded in the default results template 
![image](https://user-images.githubusercontent.com/17882290/136771594-8f0c5739-dc6a-429b-953b-333df8076be0.png)

**To Reproduce**
Search for something that doesn't have a working preview image or simply hard code an image src that doesn't exist in the template
```html
<pnp-filepreview data-preview-url="{{slot item @root.slots.PreviewUrl}}" data-preview-image-url="{{slot item @root.slots.PreviewImageUrl}}" data-theme-variant="{{JSONstringify @root.theme}}">
 <img alt='preview-image' width="120" src="{{slot item @root.slots.PreviewImageUrl}}" loading="lazy" onerror="this.onerror=null;this.src='{{@root.utils.defaultImage}}';" />
</pnp-filepreview>
```
Adding the image tag outside the pnp-filepreview tag handles the onerror correctly
Code below
```html
<pnp-filepreview data-preview-url="{{slot item @root.slots.PreviewUrl}}" data-preview-image-url="{{slot item @root.slots.PreviewImageUrl}}" data-theme-variant="{{JSONstringify @root.theme}}">
 <img alt='preview-image' width="120" src="{{slot item @root.slots.PreviewImageUrl}}" loading="lazy" onerror="this.onerror=null;this.src='{{@root.utils.defaultImage}}';" />
</pnp-filepreview>
<img alt='preview-image' width="120" src="{{slot item @root.slots.PreviewImageUrl}}" loading="lazy" onerror="this.onerror=null;this.src='{{@root.utils.defaultImage}}';" />
```
Shows this result
![image](https://user-images.githubusercontent.com/17882290/136772728-7330eb5c-17fd-4dc0-80ed-3edea2c630c9.png)
It does look like the change of src onerror is not working when inside the pnp-fileview tag. The onerror does trigger though (I put an console.log isside the onerror, and it prints)

**Expected behavior**
The defaultImage shows when image can't be loaded inside of pnp-filepreview tag

**Desktop (please complete the following information):**
 - Windows 10/Edge chromium

**Comment by FredrikEkstroem** [comment]
@FranckyC If you look at the last screenshot above, the first broken image inside the pnp-fileview tag is the original image src that doesn't exist, like 

https://xxxx.sharepoint.com/sites/xxx/_api/v2.1/sites/91e1c69c-b163-448f-8c9e-d66008deabd4,abb6e7d0-d190-46ff-9d48-d5e8a2919f8b/lists/0deb9352-f08e-4d39-94e3-f24790630a39/items/bdc2a5df-12de-46dc-aae1-4aefce1c7d4e/driveItem/thumbnails/0/large/content?preferNoRedirect=true

The second image tag added outside the pnp-fileview tag has changed the image src correctly like 

data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOsAAAB5CA.....

Try to replace the standard template code with this to reproduce
```html
<pnp-filepreview data-preview-url="{{slot item @root.slots.PreviewUrl}}" data-preview-image-url="{{slot item @root.slots.PreviewImageUrl}}" data-theme-variant="{{JSONstringify @root.theme}}">
  <img alt='preview-image' width="120" src="https://thisImageDoesntExist.png" loading="lazy" onerror="this.onerror=null;this.src='{{@root.utils.defaultImage}}';" />
</pnp-filepreview>
<img alt='preview-image' width="120" src="https://thisImageDoesntExist.png" loading="lazy" onerror="this.onerror=null;this.src='{{@root.utils.defaultImage}}';" />
```
My guess it's the line
```typescript
DOMPurify.default.sanitize(this.props.template)
```
from the file [FilePreviewComponent.tsx](https://github.com/microsoft-search/pnp-modern-search/blob/41c3aba2b2daf27018985fecb9ef313b54b8eb05/search-parts/src/components/FilePreviewComponent.tsx#L77) that removes the "this.src=" from the onerror attribute.

**Comment by wobba** [comment]
The default v4 template is the cards layout, which is different from what you are logging. Testing using the List template I see fallback to the dummy image we have - and not a broken one.

That said it seems we are not sanitizing v4 properly as `onclick` and `onerror` attributes work. They should be removed and we have to handle it specially like in v3 by providing a fallback url to the preview component.

**Comment by FredrikEkstroem** [comment]
@wobba I saw the release published was only for v3. Will it come to v4 too?

**Comment by shridha21** [comment]
Was this published for V4? I am currently using version 4.5.3.
